packages/web/src/js/__tests__/FileUploader.test.ts

```diff
@@ -0,0 +1,98 @@
+import { clearFixture, getFixture } from '../../../tests/helpers/fixture';
+import FileUploader from '../FileUploader';
+
+describe('FileUploader', () => {
+  let fixtureEl: Element;
+  let container: HTMLElement;
+  let input: HTMLInputElement;
+  let instance: FileUploader;
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    input = document.createElement('input');
+    input.type = 'file';
+    container.appendChild(input);
+    instance = new FileUploader(container);
+    fixtureEl = getFixture();
+  });
+
+  afterEach(() => {
+    instance.clearFileQueue();
+    clearFixture();
+  });
+
+  test('should be an instance of FileUploader', () => {
+    expect(instance).toBeInstanceOf(FileUploader);
+  });
+
+  test('NAME should be "fileUploader"', () => {
+    expect(FileUploader.NAME).toBe('fileUploader');
+  });
+
+  test('isDraggable should be true', () => {
+    expect(instance.isDraggable).toBeTruthy();
+  });
+
+  test('checkAllowedFileSize should return true for allowed file size', () => {
+    const file = new File([''], 'test.txt', { type: 'text/plain', lastModified: Date.now() });
+    expect(instance.checkAllowedFileSize(file)).toBeTruthy();
+  });
+
+  test('checkFileQueueDuplicity should return false for unique file name', () => {
+    const file = new File([''], 'test.txt', { type: 'text/plain', lastModified: Date.now() });
+    expect(instance.checkFileQueueDuplicity(file)).toBeFalsy();
+  });
+
+  test('clearFileQueue should remove all files from the queue', () => {
+    const file1 = new File([''], 'test1.txt', { type: 'text/plain', lastModified: Date.now() });
+    const file2 = new File([''], 'test2.txt', { type: 'text/plain', lastModified: Date.now() });
+    instance.addToQueue(file1);
+    instance.addToQueue(file2);
+    instance.clearFileQueue();
+    expect(instance.getFileQueue.size).toBe(0);
+  });
+
+  describe('constructor', () => {
+    it('should take care of element passed as a CSS selector', () => {
+      fixtureEl.innerHTML = `
+        <div class="FileUploader" data-toggle="fileUploader">
```

Comment:
this test is weird to me, won't it break If we change anything in the markup? And if not, what does it actually test? And is there actually `fileUploaderBySelector.element` (nowhere else is just `element`, but maybe its Jest thing)?

Reply:
Most tests here have this type of test. Including classes

Reply:
I think that the entire markup is not needed for this test. This tests the autoload functionality and that the right element is passed to the class. So this should fail when you change the data-toggle value or remove the data-toggle.

Reply:
done